packages/react/src/WhenVisible.ts

```diff
@@ -33,40 +33,41 @@ const WhenVisible = ({ children, data, params, buffer, as, always, fallback }: W
     }
 
     return params
-  }
+  }, [params, data])
 
   useEffect(() => {
     if (!ref.current) {
       return
     }
-
-    observer.current = new IntersectionObserver(
+    let observer: IntersectionObserver
+    observer = new IntersectionObserver(
       (entries) => {
         if (!entries[0].isIntersecting) {
           return
         }
 
-        if (!always) {
-          observer.current?.disconnect()
+        if (!always && fetchedOnce) {
```

Comment:
maybe here we need to call .current?
```suggestion
        if (!always && fetchedOnce.current) {
```

Reply:
Oops yeah my bad. Fixed it.